pilot/cmd/pilot-agent/options/agent.go

```diff
@@ -30,13 +29,13 @@ import (
 const xdsHeaderPrefix = "XDS_HEADER_"
 
 func NewAgentOptions(proxy *model.Proxy, cfg *meshconfig.ProxyConfig) *istioagent.AgentOptions {
-	proxy.DiscoverIPVersions()
+	proxy.DiscoverIPMode()
```

Comment:
why we need this again, we should do it only once in `initProxy`

Reply:
yes. This is not needed